### ISSUE TITLE
[FEA]: More options for z score calculations in DFP

### ISSUE THREAD

**Issue by raykallen** [nv-morpheus/Morpheus#497]
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

Medium

### Please provide a clear description of problem this feature solves

current DFP zscore calculations are not ideal for datasets with non-normal distributions zscores or datasets with variations between training and validation

### Describe your ideal solution

- [x] DFP to optionally use validation loss stats for zscore calculation instead just training loss stats
- [x] DFP to optionally use modified zscore (MAD)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open feature requests](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22%2Cimprovement%2Cenhancement) and have found no duplicates for this feature request

**Comment by raykallen** [comment]
closed by https://github.com/nv-morpheus/dfencoder/pull/6 (MAD) and https://github.com/hsin-c/dfencoder/pull/1 (val loss)
